From: Simon McVittie <smcv@debian.org>
Date: Thu, 26 Jul 2018 09:55:34 +0100
Subject: Skip test for GNOME#753605

It relies on a non-free JPEG that happens to exhibit the bug, which is
excluded from the Debian source package.

Forwarded: not-needed
---
 tests/meson.build   | 1 -
 tests/pixbuf-jpeg.c | 6 ++++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/tests/meson.build b/tests/meson.build
index ab7ff1e..ff941be 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -134,7 +134,6 @@ test_data = [
   'bug143608-comment.jpg',
   'bug725582-testrotate.jpg',
   'bug725582-testrotate.png',
-  'bug753605-atsize.jpg',
   'cve-2015-4491.bmp',
   'large.png',
   'large.jpg',
diff --git a/tests/pixbuf-jpeg.c b/tests/pixbuf-jpeg.c
index 3b1f2e4..ffd65f9 100644
--- a/tests/pixbuf-jpeg.c
+++ b/tests/pixbuf-jpeg.c
@@ -127,6 +127,12 @@ test_at_size (void)
       return;
     }
 
+  if (!g_file_test (g_test_get_filename (G_TEST_DIST, "bug753605-atsize.jpg", NULL), G_FILE_TEST_EXISTS))
+    {
+      g_test_skip ("non-free test data removed");
+      return;
+    }
+
   ref = gdk_pixbuf_new_from_file (g_test_get_filename (G_TEST_DIST, "bug753605-atsize.jpg", NULL), &error);
   g_assert_no_error (error);
   g_object_unref (ref);
